apm-agent-attach-cli/src/test/java/co/elastic/apm/attach/AgentDownloaderTest.java

```diff
@@ -0,0 +1,99 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.attach;
+
+import co.elastic.apm.attach.bouncycastle.BouncyCastleVerifier;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Locale;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+
+class AgentDownloaderTest {
+
+    private static final String TEST_KEY_ID = "90AD76CD56AA73A9";
+
+    private final AgentDownloader agentDownloader = new AgentDownloader(new BouncyCastleVerifier());
+
+    @Test
+    void testMavenUrl() throws Exception {
+        String mavenUrl = agentDownloader.getAgentMavenBaseUrl("1.24.0");
+        assertThat(mavenUrl).isNotNull();
+        assertThat(mavenUrl).isEqualTo("https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent/1.24.0");
+    }
+
+    @Test
+    void testWrongVersion() {
+        assertThatThrownBy(() -> agentDownloader.getAgentMavenBaseUrl("1.24.1")).isInstanceOf(IllegalArgumentException.class);
+        assertThatThrownBy(() -> agentDownloader.getAgentMavenBaseUrl("error")).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void testDownloadFile() throws Exception {
+        String agentVersion = "1.25.0";
+        String mavenAgentBaseUrl = agentDownloader.getAgentMavenBaseUrl(agentVersion);
+        String agentPgpSignatureFileName = String.format(Locale.ROOT, "%s.asc", agentDownloader.computeAgentJarName(agentVersion));
+        String mavenPgpSignatureUrl = agentDownloader.computeFileUrl(mavenAgentBaseUrl, agentPgpSignatureFileName);
+        System.out.println("mavenPgpSignatureUrl = " + mavenPgpSignatureUrl);
+        Path targetDir = Utils.getTargetAgentDir(agentVersion);
+        String targetFileName = "test.asc";
+        final Path localFilePath = targetDir.resolve(targetFileName);
+        Files.deleteIfExists(localFilePath);
+        agentDownloader.downloadFile(mavenPgpSignatureUrl, localFilePath);
+        assertThat(Files.isReadable(localFilePath)).isTrue();
+        assertThatThrownBy(() -> agentDownloader.downloadFile(mavenPgpSignatureUrl, localFilePath)).isInstanceOf(FileAlreadyExistsException.class);
+    }
+
+    // disabled due to overhead, but very convenient for sanity testing
+    @Disabled
```

Comment:
I'm not really a fan of having relevant tests disabled by default, do you think it would be possible to only disable them if they introduce test failures due to a dependency on maven central ? Ideally those kind of tests would be enabled but part of a JUnit test group that is run with retries (or for which we allow to have failures, but those aren't blocking the build).

Reply:
You're right, we download lots of stuff from maven already, this makes sense that it's enabled

Reply:
Enabling these tests